retriever/gitlabretriever/retriever.go

```diff
@@ -0,0 +1,74 @@
+package gitlabretriever
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	httpretriever "github.com/thomaspoignant/go-feature-flag/retriever/httpretriever"
+
+	"github.com/thomaspoignant/go-feature-flag/internal"
+)
+
+// Retriever is a configuration struct for a GitHub retriever.
+type Retriever struct {
+	Branch         string // default is main
+	FilePath       string
+	GitlabToken    string
+	RepositorySlug string
+	URL            string        // https://gitlab.com
```

Comment:
If I understand correctly URL is the based DNS of your gitlab instance?
If yes I am not sure if the URL is the right name here.

I would use something like GitlabBaseURL.

WDYT?